docs/client/getting-started.mdx

```diff
@@ -244,52 +291,65 @@ We recommend storing the user's addresses and creating a function like [`getAddr
+  getCosmosSigner,
+  getEvmSigner,
+  getSvmSigner,
+  onTransactionCompleted: async (txInfo) => {
     console.log(
-      `Route completed with tx hash: ${txHash} & status: ${status.state}`
+      `Route completed on chain ${txInfo.chainId} with tx hash: ${txInfo.txHash} & status: ${txInfo.status?.state}`
     );
   },
-  // called after the transaction that the user signs gets broadcast on chain
-  onTransactionBroadcast: async ({ txHash, chainID }) => {
-    console.log(`Transaction broadcasted with tx hash: ${txHash}`);
+  onTransactionBroadcast: async ({ txHash, chainId }) => {
+    console.log(`Transaction broadcasted on ${chainId} with tx hash: ${txHash}`);
   },
-  // called after the transaction that the user signs is successfully registered for tracking
-  onTransactionTracked: async ({ txHash, chainID }) => {
-    console.log(`Transaction tracked with tx hash: ${txHash}`);
+  onTransactionTracked: async ({ txHash, chainId, explorerLink }) => {
+    console.log(`Transaction tracked for ${chainId} with tx hash: ${txHash}, explorer: ${explorerLink}`);
   },
-  // called after the user signs a transaction
-  onTransactionSigned: async ({ chainID }) => {
-    console.log(`Transaction signed with chain ID: ${chainID}`);
+  onTransactionSigned: async ({ chainId }) => {
+    console.log(`Transaction signed for ${chainId}`);
   },
-  // validate gas balance on each chain
-  onValidateGasBalance: async ({ chainID, txIndex, status }) => {
-    console.log(`Validating gas balance for chain ${chainID}...`);
+  onValidateGasBalance: async (validation) => {
+    if (validation.status === "error") {
+      console.warn(`Insufficient gas balance or gas validation error on chain ${validation.chainId} (Tx Index: ${validation.txIndex}).`);
+    }
   },
+  onApproveAllowance: async (approvalInfo) => {
+    console.log(`ERC20 allowance ${approvalInfo.status} for token ${approvalInfo.allowance?.tokenContract} on chain ${approvalInfo.allowance?.chainId}`);
+  }
 });
 ```
 
-For routes that consist of multiple transactions, this will monitor each transaction until it completes, then generate the transaction for the next step and prompt the user to sign it.
+For routes that consist of multiple transactions, `executeRoute` will monitor each transaction until it completes, then generate the transaction for the next step and prompt the user to sign it using the appropriate signer.
 
 <Info>
-Alternatively, the route object allows you to generate, sign, and submit messages individually. You can also implement your own solutions for any or all of these steps. For more details on these lower-level functions, refer to `skipClient.messages`, `skipClient.signMultiChainMessageDirect`, and `skipClient.submitTransaction`.
+Alternatively, you can handle message generation, signing, and submission manually using the individual functions:
+- `messages`: Generate transaction messages.
+- `messagesDirect`: Sign messages directly
```

Comment:
messagesDirect isn't for signing message directly, it's a combination between /route and /msgs according to [this](https://docs.skip.build/go/api-reference/prod/fungible/post-v2fungiblemsgs_direct)